app/src/main/kotlin/PushYourLuck/DiceClass.kt

```diff
@@ -0,0 +1,55 @@
+package PushYourLuck
+
+import StatelessSolver
+import kotlin.random.Random
+
+class DiceClass(val nDice: Int,
+                val nSides: Int,
+                var markedSides: MutableList<MutableList<Boolean>> = Array(nDice){ i -> Array(nSides){i -> false}.toMutableList() }.toMutableList(),
+                var cumReward: Double = 0.0
+                ){
+
+
+    val diceConfig = Array(nDice){ i -> Array(nSides){i -> i + 1}.toList() }.toList()
+
+    fun roll(): MutableList<MutableList<Boolean>> {
+        // For each dice generate a random integer between (including) 0 to nSides-1
+        var diceRollResults = mutableListOf<Int>()
+        for (d in markedSides) {
+            diceRollResults.add(Random.nextInt(0, nSides-1))
+            // Use -1 to represent unmarking
+        }
+
+
+        for (d in markedSides.indices) {
+            var rollInd = diceRollResults[d]
+            if (diceRollResults.filter{it == 1}.size > 1){
+                markedSides[d].set(rollInd, false)
+            } else {
+                markedSides[d].set(rollInd, !markedSides[d][rollInd])
+            }
+
+        }
+        return markedSides
+    }
+
+    fun cashOut(): Double {
+        for (d_ind in diceConfig.indices){
+            if (markedSides[d_ind] != Array(nSides){i -> false}.toMutableList() ){
```

Comment:
I'm assuming this condition is here to make sure that the dice has at least one marked side? If so, it should use the Kotlin iterable function `any`: https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/any.html

Reply:
I've shortened the notation for this version - and provided an alternate syntactic sugar in commented line in the updated PR , but I still like this version, because its more readable.